docs/intro/overrides.rst

```diff
@@ -1,23 +1,116 @@
 .. _intro-overrides:
 
+Apply Rules
+===========
+
+Overview
+--------
+
+@handle_urls
+~~~~~~~~~~~~
+
+web-poet provides a :func:`~.handle_urls` decorator, which allows to
+declare how the page objects can be used (applied):
+
+* for which websites / URL patterns they work,
+* which data type (item classes) they can return,
+* which page objects can they replace (override; more on this later).
+
+.. code-block:: python
+
+    from web_poet import ItemPage, handle_urls
+    from my_items import MyItem
+
+    @handle_urls("example.com")
+    class MyPage(ItemPage[MyItem]):
+        # ...
+
+
+``handle_urls("example.com")`` can serve as a documentation, but it also enables
+getting the information about page objects programmatically.
+The information about all page objects decorated with
+:func:`~.handle_urls` is stored in ``web_poet.default_registry``, which is
+an instance of :class:`~.PageObjectRegistry`. In the example above the
+following :class:`~.ApplyRule` is added to the registry:
+
+.. code-block::
+
+    ApplyRule(
+        for_patterns=Patterns(include=('example.com',), exclude=(), priority=500),
+        use=<class 'MyPage'>,
+        instead_of=None,
+        to_return=<class 'MyItem'>,
+        meta={}
+    )
+
+Note how ``rule.to_return`` is set to ``MyItem`` automatically.
+This can be used by libraries like `scrapy-poet`_. For example,
+if a spider needs to extract ``MyItem`` from some page on the ``example.com``
+website, `scrapy-poet`_ now knows that ``MyPage`` page object can be used.
+
+.. _scrapy-poet: https://scrapy-poet.readthedocs.io
+
+Specifying the URL patterns
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:func:`~handle_urls` decorator uses url-matcher_ library to define the
+URL rules. Some examples:
+
+.. code-block:: python
+
+    # page object can be applied on any URL from the example.com domain,
+    # or from any of its subdomains
+    @handle_urls("example.com")
+
+    # page object can be applied on example.com pages under /products/ path
+    @handle_urls("example.com/products/")
+
+    # page object can be applied on any URL from example.com, but only if
+    # it contains "productId=..." in the query string
+    @handle_urls("example.com?productId=*")
+
+Please consult with the url-matcher_ documentation for more; it is pretty
```

Comment:
```suggestion
Note that ``@handle_urls`` automatically converts a string value into a
:class:`url_matcher.matcher.Patterns` instance when passing it in 
:class:`~.ApplyRule`'s `for_patterns` paramater
Please consult with the url-matcher_ documentation for more information; it is pretty
```

Reply:
I tried defining it as ":external:class:`url_matcher.matcher.Patterns`" but intersphinx for some reason errors out. We can define it simply as ":class:`url_matcher.matcher.Patterns`" for now to make the styling work.

Reply:
I think it might be a bit too early to talk about ApplyRule's for_patterns parameter, but you're right that it makes sense to mention that the string is converted to Patterns. Let me try to re-phrase it.

Reply:
Done in https://github.com/scrapinghub/web-poet/pull/90/commits/3e23c514bdc47010f6c274a0d1994fa160a15b6e